### PR TITLE
Use logging macros in more demos

### PR DESCRIPTION
**Pull request by dhood**
Followup of https://github.com/ros2/demos/pull/191 where a few were changed as prototypes

CI linux arm [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/881/) (still checking that the tests pass)

Needed to update expected output and add force flush for some nodes.

### CODE REVIEW

**Comment by dhood** [comment]
I'll look into why test output doesn't match correctly (probably missed some force flush of stdout) if anyone wants to review the logger usage in the meantime

File: image_tools/src/cam2image.cpp [hunk 1]
```diff
@@ -101,7 +101,7 @@ int main(int argc, char * argv[])
 
   // Initialize a ROS 2 node to publish images read from the OpenCV interface to the camera.
   auto node = rclcpp::Node::make_shared("cam2image");
-  auto node_logger = node->get_logger();
+  rclcpp::Logger node_logger = node->get_logger();
```
Comment (wjwwood) [inline image_tools/src/cam2image.cpp]: Out of curiosity, why do you think that?

Comment (dhood) [reply image_tools/src/cam2image.cpp]: in http://ci.ros2.org/job/ci_windows/3833/ it failed at the decltype check in the macro. it's just a guess though :)

Comment (wjwwood) [reply image_tools/src/cam2image.cpp]: I see, that might be, I didn't consider it was interacting with decltype, but even then it _should_ work :). Hopefully you'll find the issue.

Comment (dhood) [reply image_tools/src/cam2image.cpp]: thanks to your pointers I opened https://github.com/ros2/rclcpp/pull/423 and the usage in the lambda below can pass the type check

**Comment by dhood** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/3738/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/890/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3080/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/3833/)

windows requires https://github.com/ros2/rclcpp/pull/423. CI for windows including https://github.com/ros2/rclcpp/pull/423 [Build Status](http://ci.ros2.org/job/ci_windows/3838/)

**Comment by dhood** [comment]
more ci including https://github.com/ros2/rclcpp/pull/423
linux [Build Status](http://ci.ros2.org/job/ci_linux/3744/)
osx [Build Status](http://ci.ros2.org/job/ci_osx/3084/)

**Pull request merged**